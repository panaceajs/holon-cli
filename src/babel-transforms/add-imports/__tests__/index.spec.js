const pluginTester = require('babel-plugin-tester');
const actionCreatorsImports = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: actionCreatorsImports,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/without-imports.js',
      snapshot: false
    },
    {
      title: 'it should add action creator import.',
      fixture: '__fixtures__/without-imports.js',
      pluginOptions: {
        imports: { foo: ['first', 'second', { default: 'all' }] }
      }
    }
    // {
    //   title: 'it should add action creator import to targetPath',
    //   fixture: '__fixtures__/without-imports.js',
    //   pluginOptions: {
    //     targetPath: '../../actions',
    //     actionTypes: { foo: 'foo' }
    //   }
    // },
    // {
    //   title: 'it should update action creator import',
    //   fixture: '__fixtures__/with-imports.js',
    //   pluginOptions: {
    //     targetPath: '../actions',
    //     actionTypes: {
    //       existingAction: 'existingAction',
    //       nonExistingAction: 'nonExistingAction'
    //     }
    //   }
    // }
  ]
});
