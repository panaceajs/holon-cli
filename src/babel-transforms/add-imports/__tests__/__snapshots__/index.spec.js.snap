// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-add-import it should add action creator import.: it should add action creator import. 1`] = `
"
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

import all, { first, second } from 'foo';
console.log('nothing going on here');
"
`;
